tests/base.go

```diff
@@ -2,65 +2,56 @@ package tests
 
 import (
 	"bytes"
-	"fmt"
-	"io/ioutil"
-	"log"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
 	"net"
 	"os"
 	"os/exec"
 	"strings"
 	"time"
 
-	"github.com/goombaio/namegenerator"
+	"github.com/pkg/errors"
 	"golang.org/x/crypto/ssh"
 )
 
-// UpWg used for up wireguard
-func UpWg(wgConfig string) {
+// UpWg used for setting up the wireguard interface
+func UpWg(wgConfig string) (string, error) {
 	f, err := os.Create("/tmp/test.conf")
-
 	if err != nil {
-		log.Fatal(err)
+		return "", errors.Wrapf(err, "error creating file")
 	}
-
 	defer f.Close()
 
-	_, err2 := f.WriteString(wgConfig)
-
-	if err2 != nil {
-		log.Fatal(err2)
+	_, err = f.WriteString(wgConfig)
+	if err != nil {
+		return "", errors.Wrapf(err, "error writing wireguard config string")
 	}
 
 	cmd := exec.Command("sudo", "wg-quick", "up", "/tmp/test.conf")
 	stdout, err := cmd.Output()
-
 	if err != nil {
```

Comment:
use CombinedOutput instead of output, we are also interested in the stderr https://pkg.go.dev/os/exec#Cmd.CombinedOutput, you can also set the Stdout, Stderr fields on the cmd struct